### PR TITLE
Add `Deprecated` badge

### PR DESCRIPTION
**Pull request by ignas-k**
## Changes

- Added deprecated badge.
- Deprecated `BadgeType` (re-exported `appui-abstract` type) and added `BadgeKind` as replacement.
- Deprecated all properties that use `BadgeType` and created `badgeKind` properties that use `BadgeKind` type as replacements.
- Deprecated `PopupItem`, `PopupItemProps`, `PopupItemWithDrag`, `PopupItemWithDragProps`, `ToolbarPopupContext`, `ToolbarPopupContextProps`, `useToolbarPopupContext`. These components, context and hook are used internally to construct toolbars. I chose to deprecate these now because I used the new `BadgeKind` type inside the components but did not want to change the public props of `PopupItem` and `PopupItemWithDrag`.
- Also added internal `CommonToolbarItemWithBadgeKind`, `GroupButtonWithBadgeKind`, `ActionButtonWithBadgeKind` types which extend types from `appui-abstract` to allow the toolbars to work with the new `BadgeKind` type. Not the cleanest solution but all of this logic is internal so there are no changes to the public api concerning this change.

## Testing
Tested in standalone test-app. Added some tests for badges in various components. Updated storybook to showcase badges in toolbars, widget tabs and context menu.

### CODE REVIEW

File: ui/core-react/src/core-react/badge/DeprecatedBadge.scss [hunk 1]
```diff
@@ -2,13 +2,10 @@
...
-      &:before {
-        width: calc(100% - 1em) !important;
-        left: 0.5em !important;
-      }
-    }
+.core-badge-deprecatedBadge {
+  transform: rotate(270deg);
+
+  svg {
+    fill: rgba(0, 0, 0, 0.85);
```
Comment (ignas-k) [inline ui/core-react/src/core-react/badge/DeprecatedBadge.scss:9]: @GerardasB also do you think it is fine if I used just plain color here? I tried looking for a proper iui variable but all of them change depending on theme or are not visually correct (e.g. too light so contrast is weird). This badge specifically the cross needs to be really black for all themes to not have any weird contrast issues.

File: ui/core-react/src/core-react/badge/Badge.tsx [hunk 2]
```diff
@@ -7,22 +7,28 @@
...
 
-/** Properties for the [[Badge]] component.
+/* eslint-disable deprecation/deprecation */
+
+/** Properties for the {@link Badge} component.
  * @internal
  */
 export interface BadgeProps {
-  type?: BadgeType;
+  type?: BadgeType | BadgeKind;
```
[the remark this answers is not in the document]
Comment (ignas-k) [reply ui/core-react/src/core-react/badge/Badge.tsx:22]: Are you talking about the re-definition in `ContextMenuItemProps`?

[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/core-react/src/core-react/badge/Badge.tsx:22]: Yes

File: ui/core-react/src/core-react/badge/DeprecatedBadge.scss (hunk 1 above)
Comment (GerardasB) [reply ui/core-react/src/core-react/badge/DeprecatedBadge.scss:9]: Looks good to me

File: ui/core-react/src/core-react/badge/Badge.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ignas-k) [reply ui/core-react/src/core-react/badge/Badge.tsx]: Updated

[the remark this answers is not in the document]
Comment (ignas-k) [reply ui/core-react/src/core-react/badge/Badge.tsx:22]: I reverted the change. Even with this hack the api extractor shows an error `(ae-incompatible-release-tags) The symbol "badgeKind" is marked as @public, but its signature references "BadgeProps" which is marked as @internal`. So I suppose there is no way to work around this release tag problem. Do you think it is fine to leave the type inlined?

[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/core-react/src/core-react/badge/Badge.tsx:22]: Sure, let's keep it inlined.

**Pull request merged**